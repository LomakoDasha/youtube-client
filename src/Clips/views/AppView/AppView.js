export default class AppView {
  constructor(data) {
    this.data = data;
  }

  render() {
    const content = document.createElement('ul');
    content.className = 'main__clips';

    content.innerHTML = this.data.map(item => `<li class="main__clips__item">
      <div class="clip-inner">
        <img  class="clip-inner__image" src="${item.img}" alt="">
        <h2 class="clip-inner__title"><a class="clip-inner__link" href="https://www.youtube.com/watch?v=${item.id}" target="_blank"> ${item.title} </a></h2>
        <h3 class="clip-inner__channel"> ${item.channel} </h3>
        <h3 class="clip-inner__date"> ${item.date} </h3>
        <h3 class="clip-inner__view-count">${item.view}</h3>
        <p class="clip-inner__description"> ${item.description} </p>
      </div>
    </li>`);

    document.body.appendChild(content);
  }
}
