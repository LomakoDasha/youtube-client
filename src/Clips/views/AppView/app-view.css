body {
  margin: 0;
  padding: 0;
  background-color:â€‹#f0eadb;
}

.main__clips {
  list-style: none;
  margin-top: 10%;
  overflow-x: scroll;
  padding: 0;
  white-space: nowrap;
}

.main__clips__item {
  cursor: grab;
  overflow: hidden;
  white-space: normal;
  display: inline-block;
  width:  calc(100vw/4.9);
  height: calc(100vh/2);
  border: 5px double #8C020C;
  border-radius: 20px;
  background-color: #D6D6D6;
  margin:  calc(1em * 30 / 16);
}

@keyframes rattle {
  15% {
    transform: rotateZ(5deg);
  }
  30% {
    transform: rotateZ(3deg);
  }
  40% {
    transform: rotateZ(1px);
  }
  50% {
    transform: rotateZ(-1px);
  }
  65% {
    transform: rotateZ(-3deg);
  }
  100% {
    transform: rotateZ(-5deg);
  }
}

.main__clips__item:hover {
  animation: rattle 0.2s ease;
}

.clip-inner {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}

.clip-inner__image {
  text-align: center;
  width: 100%;
  height: 200px;
}

.clip-inner__title {
  text-align: center;
  font-family: Arial, sans-serif;
  font-size: 1.2rem;
  font-weight: 700;
  background-color: #8C020C;
  opacity: 0.8;
  margin: calc(1em * 5 / 16) calc(1em * 10 / 16);
  padding: calc(1em * 5 / 16) calc(1em * 10 / 16);
}

.clip-inner__title:hover {
  background-color: #EA7272;
  transition: 200ms ease-in-out;
}

.clip-inner__link {
  text-decoration: none;
  color: #fff;
}

.clip-inner__channel {
  background: url('http://s1.iconbird.com/ico/1212/267/w24h241355941024businessman3.png') no-repeat scroll 0 50%;
  font-family: Arial, sans-serif;
  font-size: 1rem;
  font-weight: 700;
  color: #000;
  margin: calc(1em * 5 / 16) calc(1em * 15 / 16);
  padding: calc(1em * 5 / 16) 0 calc(1em * 5 / 16) calc(1em * 60 / 16);
}

.clip-inner__date {
  background: url('http://s1.iconbird.com/ico/0912/fugue/w24h241349011746calendarmonth.png') no-repeat scroll 0 50%;
  font-family: Arial, sans-serif;
  font-size: 1rem;
  font-weight: 700;
  color: #000;
  margin: calc(1em * 5 / 16) calc(1em * 15 / 16);
  padding: calc(1em * 5 / 16) 0 calc(1em * 5 / 16) calc(1em * 60 / 16);
}

.clip-inner__view-count {
  background: url('http://s1.iconbird.com/ico/0912/fugue/w24h241349011761eye.png') no-repeat scroll 0 50%;
  font-family: Arial, sans-serif;
  font-size: 1rem;
  font-weight: 700;
  color: #000;
  margin: calc(1em * 5 / 16) calc(1em * 15 / 16);
  padding: calc(1em * 5 / 16) 0 calc(1em * 5 / 16) calc(1em * 60 / 16);
}

.clip-inner__description {
  height: calc(1em * 80 / 16);
  overflow: hidden;
  font-family: Arial, sans-serif;
  font-size: 0.9rem;
  font-weight: 400;
  color: #000;
  margin:  calc(1em * 5 / 16) calc(1em * 10 / 16);
  padding: calc(1em * 10 / 16);
}

@media only screen and (max-width: 480px) {
  .main__clips__item {
    width:  calc(100vw/1);
    margin:  calc(1em * 30 / 16);
  }
}

@media only screen and (max-width: 800px) {
  .main__clips__item {
    width:  calc(100vw/2);
    margin:  calc(1em * 30 / 16);
  }
}

@media only screen and (max-width: 1280px) {
  .main__clips__item {
    width:  calc(100vw/3.4);
    margin:  calc(1em * 30 / 16);
  }
}
