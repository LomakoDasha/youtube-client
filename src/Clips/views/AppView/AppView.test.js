import AppView from './AppView';

describe('#AppView.constructor', () => {
  it('Should be not NULL', () => {
    expect(AppView.constructor).not.toBe('');
  });
});

describe('AppView.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      data: [
        {
          id: 'vhfxdukh',
          title: 'title 1',
          img: 'http://...',
          channel: 'channel 1',
          date: '2015-10-15',
          view: '3456',
          description: 'text',
        },
        {
          id: 'vhfryjh',
          title: 'title 2',
          img: 'http://... .com',
          channel: 'channel 2',
          date: '2012-05-15',
          view: '741',
          description: 'text text',
        },
      ],
    };

    AppView.prototype.render.call(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
