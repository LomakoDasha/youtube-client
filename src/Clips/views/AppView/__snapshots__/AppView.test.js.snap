// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should be render correctly 1`] = `
"<ul class=\\"main__clips\\"><li class=\\"main__clips__item\\">
      <div class=\\"clip-inner\\">
        <img class=\\"clip-inner__image\\" src=\\"http://...\\" alt=\\"channel 1\\">
        <h2 class=\\"clip-inner__title\\"><a class=\\"clip-inner__link\\" href=\\"https://www.youtube.com/watch?v=vhfxdukh\\" target=\\"_blank\\"> title 1 </a></h2>
        <h3 class=\\"clip-inner__channel\\"> channel 1 </h3>
        <h3 class=\\"clip-inner__date\\"> 2015-10-15 </h3>
        <h3 class=\\"clip-inner__view-count\\">3456</h3>
        <p class=\\"clip-inner__description\\"> text </p>
      </div>
    </li>,<li class=\\"main__clips__item\\">
      <div class=\\"clip-inner\\">
        <img class=\\"clip-inner__image\\" src=\\"http://... .com\\" alt=\\"channel 2\\">
        <h2 class=\\"clip-inner__title\\"><a class=\\"clip-inner__link\\" href=\\"https://www.youtube.com/watch?v=vhfryjh\\" target=\\"_blank\\"> title 2 </a></h2>
        <h3 class=\\"clip-inner__channel\\"> channel 2 </h3>
        <h3 class=\\"clip-inner__date\\"> 2012-05-15 </h3>
        <h3 class=\\"clip-inner__view-count\\">741</h3>
        <p class=\\"clip-inner__description\\"> text text </p>
      </div>
    </li></ul>"
`;
